sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java

```diff
@@ -86,136 +86,111 @@ public DirectoryClient getSubDirectoryClient(String subDirectoryName) {
 
     /**
      * Creates a directory in the file share and returns a response of {@link DirectoryInfo} to interact with it.
-     *
      * <p><strong>Code Samples</strong></p>
-     *
      * <p>Create the directory</p>
-     *
      * {@codesnippet com.azure.storage.file.directoryClient.createDirectory}
-     *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory">Azure Docs</a>.</p>
      *
      * @return The {@link DirectoryInfo directory info}.
      * @throws StorageException If the directory has already existed, the parent directory does not exist or directory name is an invalid resource name.
      */
     public DirectoryInfo create() {
-        return createWithResponse(null, Context.NONE).value();
+        return createWithResponse(null, null, Context.NONE).value();
     }
 
     /**
      * Creates a directory in the file share and returns a response of DirectoryInfo to interact with it.
-     *
      * <p><strong>Code Samples</strong></p>
-     *
      * <p>Create the directory</p>
-     *
-     * {@codesnippet com.azure.storage.file.directoryClient.createWithResponse#map-Context}
-     *
+     * {@codesnippet com.azure.storage.file.directoryClient.createWithResponse#map-duration-context}
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory">Azure Docs</a>.</p>
      *
      * @param metadata Optional metadata to associate with the directory.
-     * @return A response containing the directory info and the status of creating the directory.
+     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
```

Comment:
Slight tweak to the wording.

An optional timeout which will result in a RuntimeException being raised if the operation doesn't complete before the timeout concludes.